URIs.md

```diff
@@ -0,0 +1,103 @@
+
+*Non-Requirements*
+
+- *Canonical Representation*: The protocol need not require all parties produce identical identifiers for the same resource. Canonicalization is necessary where different parties must independently produce identical representations of the same object (as when used for hash verification), and assists equality checking by allowing comparison of canonical identifiers. Certain resources, especially within the web, cannot be coerced into using a single canonical identifier. Nonetheless, the standard should minimize the space of valid identifiers for a single resource and should impose unique identifiers where possible.
+
+## Specification
+
+The standard is an extension of the common [URL standard](https://datatracker.ietf.org/doc/html/rfc1738) with support for the HTTP, HTTPS, and [IPFS](https://docs.ipfs.io/how-to/address-ipfs-on-web/#native-urls) protocols as well as the newly introduced `chain` and `farcaster` schemes.
+
+```
+<protocol>://<resource>,  <= 2048 characters
+
+protocol      =  "https" | "http" | "ipfs" | "chain" | "farcaster"
+resource      = (defined by each protocol specification)
+```
+
+The Farcaster URL structure adds the following additional protocols to the URL prefix:
+
+- `chain://`
+- `farcaster://`
+
+### Chain Resources
+
+Resources and activity across different networks will be identified by a URI with the `chain:` protocol, followed by the network's [CAIP-2](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-2.md) chain ID.
+
+Following the chain ID may be a [CAIP-10](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-10.md) account specifier, a [CAIP-19](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-19.md) asset specifier, a transaction identifier, or other unspecified identifiers for on-chain resources and activity.
+
+```
+chain://<network>
+chain://<network>:<address>
+chain://<network>/<asset_namespace>:<asset_reference>/...
+chain://<network>/tx:<tx_identifier>
+chain://<network>/<future_placeholder>
+
+network             = CAIP-2 chain id
+address             = an address string in the common format of the given network
+asset_namespace     = an identifier for a class of related assets, such as erc20 and slip44
+asset_reference     = identifies an asset within its asset_namespace
+tx_identifier       = a transaction identifier in the common format of the given network
+future_placeholder  = future message types
+...                 = addition uri parameters for the standard (optional)
+```
+
+Examples:
+
+```
+chain://eip155:1:0xaba7161a7fb69c88e16ed9f455ce62b791ee4d03
+chain://eip155:1/erc721:0xaba7161a7fb69c88e16ed9f455ce62b791ee4d03/7894
```

Comment:
it is aesthetically strange that CAIP-10 and CAIP-19 use the `:` and `/` to delimit the end of the CAIP-2 identifier. i don't think we should stray from the standard but it is odd.